nautobot/docs/plugins/region-and-site-data-migration-guide.md

```diff
@@ -0,0 +1,285 @@
+# Region and Site Related Data Model Migration Guide
+
+`Region` and `Site` models are collapsed into `Location` in Nautobot V2.0. As a result, all the `Region` and `Site` related data models are being migrated to use `Location` as well. Below is a comprehensive guide for Nautobot App developers to migrate their `Region` and `Site` related data models to `Location`.
+
+We will be using `ExampleModel` as a relatively simple and hands-on example throughout this guide to better your understanding of the migration process.
+
+## Add Location Fields to Site and Region Related Data Models
+
+`ExampleModel` right now has a `site` ForeignKey field but it does not have a `location` ForeignKey field. We need to add one before we run any other migrations in this guide.
+
+!!! note
+    You can skip this step your data models already have both `site`/`region` fields and a `location` field.
+
+```python
+# models.py
+
+class ExampleModel(OrganizationalModel):
+    site = models.ForeignKey(to="dcim.Site", on_delete=models.PROTECT, blank=True, null=True)
+    name = models.CharField(max_length=20, help_text="The name of this Example.")
+    number = models.IntegerField(default=100, help_text="The number of this Example.")
+    csv_headers = ["name", "number"]
+
+    class Meta:
+        ordering = ["name"]
```

Comment:
Not sure `number`, `csv_headers`, and the `Meta` class add much to this example - consider removing them?